File: csharp/hamming/Hamming.cs
using System;

namespace hamming
{
	public class Hamming
	{
		public static int Compute (string strandA, string strandB)
		{
			int distance = 0;
			for (int i = 0; i < strandA.Length; i++)
			{
				if (strandA[i] != strandB[i])
				{
					distance++;
				}
			}
			return distance;
		}
	}
}

File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 2

  def self.compute(strand_a, strand_b)
    raise ArgumentError unless strand_a.length == strand_b.length
    (0...strand_a.length).count do |i|
      strand_a[i] != strand_b[i]
    end
  end
end